qiime2/plugin/model/directory_format.py

```diff
@@ -118,6 +118,10 @@ def bound_path_maker(**kwargs):
             return path
         return bound_path_maker
 
+    @property
+    def path(self):
```

Comment:
Drive-by-comment on the name, we use `.path` for `FileFormat` and it is a `pathlib` object instead of a str. We might want to make that the same here. Also it would be nice if we got the file's actual path, instead of the directory format's path.

Reply:
The paths look something like this '/tmp/qiime2-test-tempmdun4tfw/file', is this not the path to the file?

Reply:
Also, the default type of `self._directory_format.path` is `qiime2.core.path.InPath`, should I just return that, or should I convert it into a `pathlib.Path`